src/shell/commands/local_partition_split.cpp

```diff
@@ -0,0 +1,752 @@
+        RETURN_FALSE_IF_NON_OK(dsn::utils::load_rjobj_from_file(rii_path, &rii),
+                               "load replica_init_info from '{}' failed",
+                               rii_path);
+
+        // x. Gather the replica.
+        to_split_partitions.push_back({replica_dir, ai, rii, pid.get_partition_index()});
+        remain_partition_ids.erase(cur_pidx);
+    }
+
+    if (!remain_partition_ids.empty()) {
+        fmt::print(stdout,
+                   "WARNING: the partitions {} are skipped to be split\n",
+                   fmt::join(remain_partition_ids, ","));
+    }
+
+    // 4. Split the partitions.
+    const auto dst_replicas_dir = dsn::utils::filesystem::path_combine(dst_data_dir, kReplicasDir);
+    auto partitions_thread_pool = std::unique_ptr<rocksdb::ThreadPool>(
+        rocksdb::NewThreadPool(static_cast<int>(lpsc.threads_per_data_dir)));
+    ddsr.psrs.reserve(to_split_partitions.size());
+    for (const auto &tsp : to_split_partitions) {
+        // Statistic the partition split result.
+        ddsr.psrs.emplace_back();
+        auto &psr = ddsr.psrs.back();
+        psr.src_replica_dir = tsp.replica_dir;
+
+        partitions_thread_pool->SubmitJob([=, &psr]() {
+            psr.success = split_partition(lpsc, tsp, dst_replicas_dir, tmp_split_replicas_dir, psr);
+        });
+    }
+    partitions_thread_pool->WaitForJobsAndJoinAllThreads();
+    if (std::any_of(ddsr.psrs.begin(), ddsr.psrs.end(), [](const PartitionSplitResult &psr) {
+            return !psr.success;
+        })) {
+        return false;
+    }
+    return true;
+}
+
+bool local_partition_split(command_executor *e, shell_context *sc, arguments args)
+{
+    // 1. Parse parameters.
+    argh::parser cmd(args.argc, args.argv);
+    RETURN_FALSE_IF_NOT(cmd.pos_args().size() >= 8,
+                        "invalid command, should be in the form of '{}'",
+                        local_partition_split_help);
+    int param_index = 1;
+    LocalPartitionSplitContext lpsc;
+    PARSE_STRS(lpsc.src_data_dirs);
+    PARSE_STRS(lpsc.dst_data_dirs);
+    PARSE_UINT(lpsc.src_app_id);
+    PARSE_UINT(lpsc.dst_app_id);
+    PARSE_UINTS(lpsc.src_partition_ids);
+    PARSE_UINT(lpsc.src_partition_count);
+    PARSE_UINT(lpsc.dst_partition_count);
+    lpsc.dst_app_name = cmd(param_index++).str();
+    PARSE_OPT_UINT("threads_per_data_dir", lpsc.threads_per_data_dir, 1);
+    PARSE_OPT_UINT("threads_per_partition", lpsc.threads_per_partition, 1);
+    lpsc.post_full_compact = cmd["--post_full_compact"];
+    lpsc.post_count = cmd["--post_count"];
+
+    // 2. Check parameters.
+    if (!validate_parameters(lpsc)) {
+        return false;
+    }
+
+    // 3. Split each data directory.
+    auto data_dirs_thread_pool = std::unique_ptr<rocksdb::ThreadPool>(
+        rocksdb::NewThreadPool(static_cast<int>(lpsc.src_data_dirs.size())));
+    CHECK_EQ(lpsc.src_data_dirs.size(), lpsc.dst_data_dirs.size());
```

Comment:
`validate_parameters` already check it

Reply:
Yes, we will use an index to iterate the 2 containers so we use CHECK_XX here to ensure the validate_parameters() works, to prevent one day the logic in validate_parameters() is removed or incorrectly modified.